### ISSUE TITLE
Feature Request: Disallow top-level element selectors

### ISSUE THREAD

**Issue by hankthewhale** [stylelint/stylelint#2255]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Feature request (I think)

A linter that disallows writing styles directly on element selectors ('h1, 'a', 'span' etc) without being wrapped in a class. Something like `selector-no-global"`

Using css-modules, the goal is to avoid writing styles directly on an element or global selector.
e.g.
```css
h1 {
  styles
}
```

and instead write your css in a component file

e.g. (in Header.css)
```css
.darkHeader {
  // styles
}

.lightHeader {
  // styles
}
```

In our project, we have a base.css file that does apply general styling to elements but once we write css for our components, we want to rely entirely on css modules to handle the class name spacing to avoid collisions with other styles. Unfortunately, there have been times where team members have written something inside of a Component.css file that leaks out to the rest of the application.

e.g.
```css
label {
  font-size: 1.5rem;
}
```

It would be nice if we could have the linter check for those cases. Then we could use the ignore line in the files that we do allow global element selectors.

I think this is a feature request because I didn't see anything in the docs that seemed to do what I'm asking for but please let me know if I overlooked anything. 

Basically to sum up what I'm thinking...

This would fail
```css
h1 {
  // styles
}
```

This would pass
```css
.someClass h1 {
  // styles
}
```

> Which rule, if any, is this issue related to?

Potentially none

> What CSS is needed to reproduce this issue?

```css
h1 .foo {
  color: pink;
}
```

> What stylelint configuration is needed to reproduce this issue?

N/A

> Which version of stylelint are you using?

`7.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`npm run stylelint \"client/**/*.css\"`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

css modules

> What did you expect to happen?

N/A

> What actually happened (e.g. what warnings or errors you are getting)?

N/A

**Comment by m-allanson** [comment]
I think this is covered by the 'selector-no-type' rule with the 'ignore: [
"descendant"]' option: http://stylelint.io/user-guide
/rules/selector-no-type/

**Comment by hankthewhale** [comment]
Ah yes! I completely overlooked this. Thanks for pointing it out @m-allanson. Closing.

**Comment by hankthewhale** [comment]
@m-allanson I think I found some undesirable behavior with this rule. Opened #2256

**Comment by japboy** [comment]
it seems like there is no `selector-no-type` any more. are there any alternative which restrict to use type selector?

**Comment by hudochenkov** [comment]
@japboy https://stylelint.io/user-guide/rules/selector-max-type/ with a value `0`.

**Comment by Eugene-Musika** [comment]
> @japboy https://stylelint.io/user-guide/rules/selector-max-type/ with a value `0`.

It seems that this rule does not cover cases as:
- top level attributes selectors `[some-attr] { height:  10px }`
- css modules selectors `:global` and `:local`

**Comment by yairEO** [comment]
I also want to disallow top-level SASS selectors which are starting with `&`

```css
&.foo { }
```

SASS

**Comment by Mouvedia** [comment]
@yairEO sounds like a job for https://github.com/stylelint-scss/stylelint-scss

**Comment by yairEO** [comment]
@Mouvedia - 
No all CSS pre-processors are SASS. Also, the `&` is coming to native CSS. This is more of top-level selector `regex` thing than SASS-specific

**Comment by jeddy3** [comment]
@yairEO Please open a new [feature request issue](https://github.com/stylelint/stylelint/issues/new?assignees=&labels=&template=REQUEST_A_FEATURE.yml) so that we can discuss.

**Comment by ekilah** [comment]
Update: I made

- https://github.com/stylelint/stylelint/issues/8406
